File: android-compiler/src/main/java/com/pranav/lib_android/code/formatter/GoogleJavaFormatter.java
package com.pranav.lib_android.code.formatter;

import com.google.googlejavaformat.java.Formatter;
import com.google.googlejavaformat.java.FormatterException;
import com.google.googlejavaformat.java.JavaFormatterOptions;

public class GoogleJavaFormatter {

    private String source;

    public GoogleJavaFormatter(String source) {
        this.source = source;
    }

    public String format() {
        var options =
                JavaFormatterOptions.builder()
                        .style(JavaFormatterOptions.Style.AOSP) // Use AOSP formatting style
                        .formatJavadoc(true) // Format Javadoc with code
                        .build();
        var formatter = new Formatter(options);
        try {
            return formatter.formatSourceAndFixImports(source);
        } catch (FormatterException e) {
            throw new IllegalStateException(e);
        }
    }
}

File: android-compiler/src/main/java/org/cosmic/ide/android/code/formatter/GoogleJavaFormatter.kt
package org.cosmic.ide.android.code.formatter

import com.google.googlejavaformat.java.Formatter
import com.google.googlejavaformat.java.FormatterException
import com.google.googlejavaformat.java.JavaFormatterOptions

class GoogleJavaFormatter(input: String) {

    private val source: String

    init {
        source = input
    }

    fun format(): String {
        val options =
                JavaFormatterOptions.builder()
                        .style(JavaFormatterOptions.Style.AOSP) // Use 4 spaces for tab
                        .formatJavadoc(true) // Format Javadoc
                        .build()
        val formatter = Formatter(options)
        try {
            return formatter.formatSourceAndFixImports(source)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return source
    }
}
